File: backend/OlivePlatform/OlivePlatform.Application/Features/Plots/Responses/PlotForListResponse.cs
namespace OlivePlatform.Application.Features.Plots.Responses;

public class PlotForListResponse
{
    public int Total { get; set; }

    public int Id { get; set; }

    public string Name { get; set; } = string.Empty;

    public string Reference { get; set; }

    public decimal AreaHectares { get; set; }

    public int NumberOfTrees { get; set; }

    public int PlantingYear { get; set; }

    public string Location { get; set; }

    public string? Notes { get; set; }

    public DateTime CreatedAt { get; set; }

    public List<PlotVarietyResponse> Varieties { get; set; } = [];
}

File: frontend/olive-platform/src/features/plots/data/responses/PlotForListResponse.ts
export interface PlotForListResponse {
  total: number;
  id: number;
  reference: string;
  name: string;
  numberOfTrees: number;
  harvestedTreesPercentage: number;
  canLaunchHarvest: boolean;
  plannedTreesPercentage : number;

}